### ISSUE TITLE
Update for GHC 8

### ISSUE THREAD

**Issue by ekmett** [ekmett/either#44]
This is currently blocked on

https://github.com/basvandijk/monad-control/issues/32

**Comment by RyanGlScott** [comment]
`monad-control`'s upper version bounds for `transformers-compat` [have been relaxed](http://hackage.haskell.org/package/monad-control-1.0.0.4/revisions/) on Hackage.

**Comment by GeorgeCo** [comment]
Stiil can't install either:

 cabal install either
cabal install either
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: either-4.4.1 (user goal)
trying: base-4.9.0.0/installed-4.9... (dependency of either-4.4.1)
next goal: transformers-base (dependency of either-4.4.1)
rejecting: transformers-base-0.4.4/installed-9bo... (conflict:
transformers-base => transformers==0.5.1.0/installed-0.5..., either =>
transformers>=0.2 && <0.5)
trying: transformers-base-0.4.4
next goal: transformers-compat (dependency of transformers-base-0.4.4)
rejecting: transformers-compat-0.5.1.4/installed-1bS... (conflict:
transformers-compat => transformers==0.5.1.0/installed-0.5..., either =>
transformers>=0.2 && <0.5)
trying: transformers-compat-0.5.1.4
trying: transformers-compat-0.5.1.4:-two
trying: transformers-compat-0.5.1.4:+mtl
trying: transformers-compat-0.5.1.4:-three
next goal: transformers (dependency of either-4.4.1)
rejecting: transformers-0.5.1.0/installed-0.5..., transformers-0.5.2.0,
transformers-0.5.1.0, transformers-0.5.0.1, transformers-0.5.0.0 (conflict:
either => transformers>=0.2 && <0.5)
rejecting: transformers-0.4.3.0, transformers-0.4.2.0 (conflict:
base==4.9.0.0/installed-4.9..., transformers => base>=2 && <4.9)
rejecting: transformers-0.4.1.0 (conflict: base==4.9.0.0/installed-4.9...,
transformers => base>=2 && <4.8 || >=1.0 && <2)
rejecting: transformers-0.3.0.0 (conflict: transformers-compat-0.5.1.4:three
=> transformers>=0.2 && <0.3 || >=0.4.1 && <0.6)
rejecting: transformers-0.2.2.1, transformers-0.2.1.0, transformers-0.2.0.0
(conflict: transformers-compat-0.5.1.4:three => transformers>=0.4.1 && <0.6)
rejecting: transformers-0.1.4.0, transformers-0.1.3.0, transformers-0.1.1.0,
transformers-0.1.0.1, transformers-0.0.1.0, transformers-0.0.0.0,
transformers-0.5.0.2 (conflict: either => transformers>=0.2 && <0.5)
rejecting: transformers-0.4.0.0 (conflict: transformers-compat =>
transformers>=0.3 && <0.4 || >=0.2 && <0.3 || >=0.4.1 && <0.6)
rejecting: transformers-0.2.2.0 (conflict: transformers-compat-0.5.1.4:three
=> transformers>=0.4.1 && <0.6)
rejecting: transformers-0.1.0.0 (conflict: either => transformers>=0.2 &&
<0.5)
Dependency tree exhaustively searched.

ghc --version
ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.0.0.20160204

**Comment by ekmett** [comment]
Have you tun `cabal update` lately since January 24th?

**Comment by RyanGlScott** [comment]
The Hackage version of `either` is out-of-sync with the repo. There's a couple of GHC 8.0 fixes that I pushed recently (bumping the upper version bounds of `transformers` and adding an explicit `forall` since 8.0 is pickier about that).

**Comment by ekmett** [comment]
So we need a new release?

**Comment by RyanGlScott** [comment]
Yes.

**Comment by nikita-volkov** [comment]
Any updates on this? The published version has an outdated `transformers (>=0.2 && <0.5)` constraint.

**Comment by mantkiew** [comment]
Doing `cabal install either --allow-newer` reveals:

```
[1 of 3] Compiling Data.Either.Validation ( src/Data/Either/Validation.hs, dist/build/Data/Either/Validation.o )

src/Data/Either/Validation.hs:120:32: error:
    Not in scope: type variable ‘p’

src/Data/Either/Validation.hs:120:43: error:
    Not in scope: type variable ‘f’

src/Data/Either/Validation.hs:120:49: error:
    Not in scope: type variable ‘p’

src/Data/Either/Validation.hs:120:54: error:
    Not in scope: type variable ‘f’

src/Data/Either/Validation.hs:120:62: error:
    Not in scope: type variable ‘p’

src/Data/Either/Validation.hs:120:67: error:
    Not in scope: type variable ‘f’
```

**Comment by mitchty** [comment]
Will the fix @mantkiew noted be pushed out to a new release of Either? As it stands this prevents bootstrapping of stack via cabal on ghc 8.0 currently due to the issue noted.

**Comment by ekmett** [comment]
I'll push out a new version.

**Comment by RyanGlScott** [comment]
[`either-4.4.1.1`](http://hackage.haskell.org/package/either-4.4.1.1) uploaded to Hackage.
